File: HW/HW10/hw10.py
from typing import Optional, List
import queue
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
    q = queue.Queue()
    result = []
    if  root is None:
        return None
    q.put(root)
    while q.qsize() != 0:
        level=[]
        queueSize = q.qsize()
        for i in range(queueSize):
            node=q.get()
            level.append(node.val)
            if node.left is not None:
                q.put(node.left)
            if node.right is not None:
                q.put(node.right)
        result.append(level)
    return result


node1 = TreeNode(1)
node3 = TreeNode(3, left=node1)
node5 = TreeNode(5)
node9 = TreeNode(9)
node8 = TreeNode(8, left=node5, right=node9)
root = TreeNode(4, left=node3, right=node8)

print(levelOrder(None, root))

File: HW/HW10/HW10.java
package hw10;
import java.util.*;

public class HW10 {
	Queue <TreeNode> queue = new LinkedList<>();
	List<List<Integer>> result = new ArrayList<>();
	public List<List<Integer>> levelOrder(TreeNode root) {
	       // Your code here
		if(root == null) return null;
		queue.add(root);
		while(!queue.isEmpty()) {
			ArrayList<Integer> level = new ArrayList<Integer>();
			int queueSize = queue.size();
			for (int i = 0; i < queueSize; i++) {
				TreeNode node = queue.poll();
				level.add(node.val);
				if (node.left != null)
					queue.add(node.left);
				if (node.right != null)
					queue.add(node.right);
				
			}
			result.add(level);

		}
		
		return result;
			
			 
	    }

	
	public static void main(String[] args) {
        TreeNode node1 = new TreeNode(1);
        TreeNode node3 = new TreeNode(3, node1, null);
        TreeNode node5 = new TreeNode(5);
        TreeNode node9 = new TreeNode(9);
        TreeNode node8 = new TreeNode(8, node5, node9);
        TreeNode root = new TreeNode(4, node3, node8);
        
        System.out.println(new HW10().levelOrder(root));

    }
}
